src/Algebra/Graph/AdjacencyIntMap/Algorithm.hs

```diff
@@ -300,6 +304,162 @@ topSort g = runContT (evalStateT (topSort' g) initialState) id where
 isAcyclic :: AdjacencyIntMap -> Bool
 isAcyclic = isRight . topSort
 
+-- TODO: Benchmark and optimise.
+-- | Compute the /condensation/ of a graph, where each vertex corresponds to a
+-- /strongly-connected component/ of the original graph. Note that component
+-- graphs are non-empty, and are therefore of type
+-- "Algebra.Graph.NonEmpty.AdjacencyMap".
```

Comment:
Should this be `AdjacencyIntMap`?

Reply:
In fact, I'm confused: are the inner graphs non-empty?

Reply:
Answering my own question: we don't seem to have an data type for non-empty adjacency maps with `Int` vertices, so I guess this requires further thinking.

Reply:
There are three options: 

1. Keep `scc` only in `AdjacencyMap`: this is easy but we lose some specialised code.
2. Make `scc` in `AdjacencyIntMap` have a different type: this is also easy, but is a bit awkward and probably not want we want in the long term.
3. Add a new module `NonEmpty.AdjacencyIntMap` and the corresponding `scc`. This is a lot of work but seems to be the right solution in the long term.

@jitwit How about we go for option (1) right now, but then you attempt (3) in a separate PR?

Reply:
Yeah, I had considered opening a pr to add the analogous `NonEmpty.AdjacencyIntMap` so that the types can match, but held off. (1) then (3) makes sense to me